source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -1509,36 +1532,48 @@ export class AdaptiveCardPeer extends TypedCardElementPeer<Adaptive.AdaptiveCard
         return true;
     }
 
-    protected internalAddCommands(commands: Array<PeerCommand>) {
-        super.internalAddCommands(commands);
+    protected internalAddCommands(hostContainer: HostContainer, targetVersion: Adaptive.Version, commands: Array<PeerCommand>) {
+        super.internalAddCommands(hostContainer, targetVersion, commands);
 
-        commands.push(
-            new PeerCommand(
-                {
-                    name: "Add an action",
-                    iconClass: "acd-icon-bolt",
-                    execute: (command: PeerCommand, clickedElement: HTMLElement) => {
-                        let popupMenu = new Controls.PopupMenu();
+        let availableActions: Adaptive.ITypeRegistration<Adaptive.Action>[] = [];
 
-                        for (var i = 0; i < Adaptive.AdaptiveCard.actionTypeRegistry.getItemCount(); i++) {
-                            let menuItem = new Controls.DropDownItem(i.toString(), Adaptive.AdaptiveCard.actionTypeRegistry.getItemAt(i).typeName);
-                            menuItem.onClick = (clickedItem: Controls.DropDownItem) => {
-                                let registryItem = Adaptive.AdaptiveCard.actionTypeRegistry.getItemAt(Number.parseInt(clickedItem.key));
-                                let action = registryItem.createInstance();
-                                action.title = registryItem.typeName;
+        for (var i = 0; i < hostContainer.actionsRegistry.getItemCount(); i++) {
+            let typeRegistration = hostContainer.actionsRegistry.getItemAt(i);
 
-                                this.addAction(action);
+            if (typeRegistration.schemaVersion.compareTo(targetVersion) <= 0) {
+                availableActions.push(typeRegistration);
+            }
+        }
 
-                                popupMenu.closePopup();
-                            };
+        if (availableActions.length > 0) {
+            commands.push(
+                new PeerCommand(
+                    {
+                        name: "Add an action",
+                        iconClass: "acd-icon-bolt",
+                        execute: (command: PeerCommand, clickedElement: HTMLElement) => {
+                            let popupMenu = new Controls.PopupMenu();
 
-                            popupMenu.items.add(menuItem);
-                        }
+                            for (let typeRegistration of availableActions) {
+                                let menuItem = new Controls.DropDownItem(i.toString(), typeRegistration.typeName);
```

Comment:
where did i come from?

Reply:
Not sure I understand the question?

Reply:
it's my fault. i should have set the comment at the exact spot. 

---
In reply to: [333671882](https://github.com/microsoft/AdaptiveCards/pull/3508#discussion_r333671882) [](ancestors = 333671882)

Reply:
Fixed with https://github.com/microsoft/AdaptiveCards/pull/3508/commits/3d3a1b8338432d304982196d0c5d7f5f77c2b59a